scipy/sparse/coo.py

```diff
@@ -289,6 +289,22 @@ def transpose(self, axes=None, copy=False):
 
     transpose.__doc__ = spmatrix.transpose.__doc__
 
+    def resize(self, shape):
+        if not isshape(shape, nonneg=True):
+            raise TypeError("shape must be a 2-tuple of positive integers")
+        new_M, new_N = shape
+        M, N = self.shape
+
+        if new_M < M or new_N < N:
+            mask = np.logical_and(self.row < new_M, self.col < new_N)
```

Comment:
Could also try checking `self.row.max() < M or self.col.max() < N` as a last effort before actually doing the indexing. Might want to look at some benchmarks to decide if it's worth the extra check.

Reply:
I might just add `if not mask.all()`.

Reply:
I.e. it may avoids the 3x fancy indexing for the cost of a fast operation. I think you're right that handling this case is useful when resizing has been used to facilitate initialising a matrix with extra capacity, then resizing after inserting data (not that you'd ordinarily use COO for that).